PhysicsTools/PatAlgos/python/slimming/customizeMiniAOD_MuEGFixMoriond2017.py

```diff
@@ -0,0 +1,138 @@
+import FWCore.ParameterSet.Config as cms
+
+from PhysicsTools.PatAlgos.tools.helpers import MassSearchReplaceAnyInputTagVisitor, cloneProcessingSnippet, addKeepStatement
+from PhysicsTools.PatAlgos.slimming.extraJets_MuEGFixMoriond2017 import backupJetsFirstStep, backupJetsSecondStep
+from RecoEgamma.EgammaTools.egammaGainSwitchFixToolsForPAT_cff import customizeGSFixForPAT
+
+def addBadMuonFilters(process):
+    process.load("RecoMET.METFilters.badGlobalMuonTaggersAOD_cff")
+    process.Flag_noBadMuons = cms.Path(process.noBadGlobalMuons)
+    process.Flag_badMuons = cms.Path(process.badGlobalMuonTagger)
+    process.Flag_duplicateMuons = cms.Path(process.cloneGlobalMuonTagger)
+    for P in process.Flag_noBadMuons, process.Flag_badMuons, process.Flag_duplicateMuons:
+        process.schedule.insert(0, P)
+
+def cleanPFCandidates(process, badMuons, verbose=False):
+    process.load("CommonTools.ParticleFlow.muonsCleaned_cfi")
+    process.patMuons.userData.userInts.src = [ cms.InputTag("muonsCleaned:oldPF") ]
+
+    process.load("CommonTools.ParticleFlow.pfCandidatesBadMuonsCleaned_cfi")
+    process.muonsCleaned.badmuons = badMuons
+    replaceMuons = MassSearchReplaceAnyInputTagVisitor("muons", "muonsCleaned", verbose=verbose)
+    needOriginalMuons = [ process.muonsCleaned ] + [ getattr(process,l.moduleLabel) for l in badMuons ]
+    replacePFCandidates = MassSearchReplaceAnyInputTagVisitor("particleFlow", "pfCandidatesBadMuonsCleaned", verbose=verbose)
+    for everywhere in [ process.producers, process.filters, process.analyzers, process.psets, process.vpsets ]:
+        for name,obj in everywhere.iteritems():
+            if obj not in needOriginalMuons:
+                replaceMuons.doIt(obj, name)
+            if obj != process.pfCandidatesBadMuonsCleaned: 
+                replacePFCandidates.doIt(obj, name)
+            
+    process.load("CommonTools.ParticleFlow.pfEGammaToCandidateRemapper_cfi")
+    process.pfEGammaToCandidateRemapper.pf2pf = cms.InputTag("pfCandidatesBadMuonsCleaned")
+    process.reducedEgamma.gsfElectronsPFValMap = cms.InputTag("pfEGammaToCandidateRemapper","electrons")
+    process.reducedEgamma.photonsPFValMap      = cms.InputTag("pfEGammaToCandidateRemapper","photons")
+    if hasattr(process,"gedGsfElectronsGSFixed"):
```

Comment:
there is no gedGsfElectronsGSFixed in 90X.
It's worth to print a warning that it's not here or remove this part completely.
IMO, we need a part of workflow that allows to redo the egamma objects for whatever reason and propagate this accordingly similar to what was done in 80X.
In that case, the fixed collection will probably have no "GS", just ```gedGsfElectronsFixed```